c7n/filters/iamaccess.py

```diff
@@ -214,7 +227,21 @@ def check_cross_account(policy_text, allowed_accounts, everyone_only,
                 principal_ok = True
 
         if 'IpAddress' in s['Condition']:
-            principal_ok = True
+            sources = s['Condition']['IpAddress']['aws:SourceIp']
+            ips = []
+            if isinstance(sources, list):
+                for src in sources:
+                    ips.append(parse_cidr(src))
+            else:
+                ips.append(parse_cidr(sources))
+
+            whitelist_ips = filter(
+                None, [parse_cidr(a) for a in allowed_accounts])
+
+            if any(i not in ip for ip in ips for i in whitelist_ips):
```

Comment:
this seems backwards, your comparing the `account` is in the ip condition, typically you'd be checking an allowed cidr =/in the configured. else you have a gap around the ip range your not checking, i.e. the unit test for this checks if a single ip is in /8 address space, instead of checking the /8 address space.

Reply:
ugh. i went back to this a few times because i thought it was backwards, but when the test passed i told myself it was right.